packages/experimental/kai/src/containers/ProjectList.tsx

```diff
@@ -66,9 +66,9 @@ export const ProjectItem: FC<{ project: Project }> = withReactor(({ project }) =
 
   // TODO(burdon): Pass in Task1, Task2.
   const handleDrag = (active: number, over: number) => {
-    const task1 = project.tasks[active];
-    const task2 = project.tasks[over];
-    project.tasks.splice(project.tasks.indexOf(task2), 0, task1);
+    const task = project.tasks[active];
+    project.tasks.splice(active, 1);
+    project.tasks.splice(over, 0, task);
```

Comment:
will this be atomic? should you use your new batch API?

Reply:
won't be atomic. I think we should have some sort of deffered batching by default. Same as react batches state updates automatically.

All mutations that follow one another should go into one batch.

This, together with cross-item batches (needed to achieve strong-reference consistency), we should do this on the database level.